raiden/transfer/token_network.py

```diff
@@ -30,64 +30,90 @@
 
+
+def subdispatch_to_channels(
+    block_hash: BlockHash,
+    block_number: BlockNumber,
+    channel_states: List[NettingChannelState],
+    state_change: StateChangeWithChannelID,
+    token_network_state: TokenNetworkState
+) -> TransitionResult:
+    events = []
     ids_to_channels = token_network_state.channelidentifiers_to_channels
+    for channel_state in channel_states:
+        channel_result = channel.state_transition(
+            channel_state=channel_state,
+            state_change=state_change,
+            block_number=block_number,
+            block_hash=block_hash,
+        )
 
-    channel_state = None
-    if node_address is not None:
-        if node_address in ids_to_channels:
-            channel_state = ids_to_channels[node_address].get(state_change.channel_identifier)
+        partner_to_channelids = token_network_state.partneraddresses_to_channelidentifiers[
+            channel_state.partner_state.address
+        ]
+
+        channel_identifier = state_change.channel_identifier
+        if channel_result.new_state is None:
+            del ids_to_channels[channel_state.our_state.address][channel_identifier]
+            partner_to_channelids.remove(channel_identifier)
         else:
-            lc_address = views.get_lc_address_by_channel_id_and_partner(token_network_state, node_address,
-                                                                    state_change.canonical_identifier)
-            node_address = lc_address
-            if lc_address in token_network_state.channelidentifiers_to_channels:
-                channel_state = token_network_state.channelidentifiers_to_channels[lc_address].get(
-                    state_change.canonical_identifier.channel_identifier)
-        if channel_state:
-            result = channel.state_transition(
-                channel_state=channel_state,
-                state_change=state_change,
-                block_number=block_number,
-                block_hash=block_hash,
-            )
-
-            partner_to_channelids = token_network_state.partneraddresses_to_channelidentifiers[
-                channel_state.partner_state.address
-            ]
+            ids_to_channels[channel_state.our_state.address][channel_identifier] = channel_result.new_state
 
-            channel_identifier = state_change.channel_identifier
-            if result.new_state is None:
-                del ids_to_channels[node_address][channel_identifier]
-                partner_to_channelids.remove(channel_identifier)
-            else:
-                ids_to_channels[node_address][channel_identifier] = result.new_state
+        events.extend(channel_result.events)
+    return TransitionResult(token_network_state, events)
 
-            events.extend(result.events)
 
-    return TransitionResult(token_network_state, events)
+def get_channels_to_dispatch_statechange(
```

Comment:
I think this function does what we need, but both the name andthe StateChangeWithChannelID doesnt. 

`StateChangeWithChannelID ` represents: "a set of state changes that must be subdispatched to two channels because the channels are from two onboarded light clients"

`get_channels_to_dispatch_statechange` logic is strongly related with two light clients of the hub, it basically return 1 or 2 channel states depending on `both_participants_are_light_clients` value. Basically, given an address and a channel id, we check if that channel is also into the state of another LC

I suggest to change StateChangeWithChannelID for MultipleChannelDispatchStateChange or something. 

About the function, the name is ok but we need docstring

Reply:
I Agree with @marcosmartinez7 here, i still don't like all this logic just to add 1 or 2 participant channel states to the list but i don't see any other way to do it, anyway adding a docstring and changing the state change name should help us to understand if we come back here later.

Reply:
StateChangeWithChannelID was introduced a long time ago. It represents state changes that hold a channel identifier. I thik MultipleChannelDispatchStateChange would lost the meaning of the type